### PR TITLE
Create convert timezone option 

### PR DESCRIPTION
**Pull request by Frydafly**
# Pull Request for dbt-arc-functions

This PR updates the actionkit email model so that the publish date timestamp converts from UTC to PST. Given that it does this using a macro from calogica/dbt_date, I've also updated first profiles fixer so that it adds this package to dbt projects and also adds the language needed in the dbt profiles yaml so that it knows what timezone to convert to. Right now, this is somewhat of a placeholder for a more robust system in the future. 

Follow up items I'd like: 
1. give user an option to choose what timezone they'd like the project to convert to when they run the first profiles fixer, right now I just default to UTC and don't ask the user for anything
2. start to incorporate this macro in places other than actionkit. I wish I could incorporate it further now but a) any models using global message summary does not have a timestamp but a date, so we can't really timezone convert and b) the models that use CRMs like everyaction would break right now since not all clients have the correct configuration for calogica as a package. So that would be nice to update and configure moving forward. and do so in a way that updates all applicable clients carefully. 

^^ I'll go ahead and make these jira tickets too so that we can keep track

cc @dmbluestate @jacksteadman @ryantimjohn 

- [not applicable, only updated a macro ] If you've created a macro, make sure it is documented by running the notebook: `process_model_documentation_codegen_output.ipynb` util in this repo

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.
tested on MercyCorps

- [not applicable] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies

- [x ] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page
Screenshot 2023-03-15 at 11 34 47 AM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this
Screenshot 2023-03-15 at 11 35 36 AM

- [x] If exists, link bug issue and jira tickets to PR
https://bluestate.atlassian.net/browse/ARC-1227?atlOrigin=eyJpIjoiNDc5ZTFjMDRiODk1NGU0ZTkxOTk2MjA1NjMzNzYwZDYiLCJwIjoiaiJ9

### CODE REVIEW

**Comment by Frydafly** [comment]
jira ticket: https://bluestate.atlassian.net/browse/ARC-1347

File: macros/frakture_actionkit_email/staging/stg_frakture_actionkit_email_jobs.sql [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
-    SAFE_CAST(NULL AS TIMESTAMP) AS delivered_timestamp,
-    SAFE_CAST(label AS STRING) AS email_name,
-    SAFE_CAST(subject AS STRING) AS email_subject, 
-    SAFE_CAST(final_primary_source_code AS STRING) as source_code
+
+with casting as (
+SELECT DISTINCT SAFE_CAST(message_id AS STRING) as message_id, 
+    SAFE_CAST(from_name AS STRING) as from_name,
+    SAFE_CAST(from_email AS STRING) as from_email,
+    SAFE_CAST(publish_date AS TIMESTAMP) as publish_date,
```
Comment (ryantimjohn) [inline macros/frakture_actionkit_email/staging/stg_frakture_actionkit_email_jobs.sql:8]: @Frydafly this seems like a roundabout way to do this, why not just do:
COALESCE({{ dbt_date.convert_timezone(SAFE_CAST(publish_date AS TIMESTAMP)) }},
                                                                         SAFE_CAST(publish_date AS TIMESTAMP)
) AS best_guess_timestamp,

In line 6 so you don't have to do the whole CTE statement? It seems to be adding a lot of code.

Willing to pass this as it seems like dbt style guides encourage CTEs:
https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md
https://docs.getdbt.com/terms/cte

Comment (ryantimjohn) [reply macros/frakture_actionkit_email/staging/stg_frakture_actionkit_email_jobs.sql:8]: I don't know that I understand how timezones for timestamps work in bigquery:
https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timezone_definitions

**Review by ryantimjohn — APPROVED** [review]
This looks fine. Would you mind checking that reference I sent and make sure that it makes sense to give a timestamp a timezone? I feel like the timezone conversion might make more sense at a later point in the pipeline (like as a step right before presentation) as a TIMESTAMP is a definite point in time independent of timezone but I'm actually not sure what the logic of your situation is. Happy to talk through!

**Comment by Frydafly** [comment]
hiya! when I first tried coalesce, the macro caused an error in my model. i
would like to try again to see if it was a fluke of some sort. so that's
the reason i turned to this extended cte version

**Comment by Frydafly** [comment]
hi! so, open to doing this at a later point for sure, the purpose of this
change was to make it possible for a client to switch over from UTC to PST,
and I wanted to do something that would let us continue to use dbt arc
functions and allow other clients to also choose a time zone if they wanted
something other than UTC. if you want to find something on my calendar, we
can also talk this one through!

also the reason I converted the publish_date to timestamp is because I
would continue to get an error while the field was datetime.

**Comment by Frydafly** [comment]
@ryantimjohn update: I did manage to remove the CTE and include 'cast(publish_date) as timestamp' inside of the macro and it processed and ran. 

*I have also applied the same process to the global transactions table since it is a table with timestamps.* In order to avoid breaking any existing dbt projects, I am going to avoid merging this branch until I have installed calogica/dbt_date package into all existing arc projects.

**Comment by ryantimjohn** [comment]
@Frydafly makes sense!

**Comment by Frydafly** [comment]
Ready to merge

**Pull request merged**